### PR TITLE
S23 add to calendar for schedule panel

### PR DESCRIPTION
**Pull request by mla04762**
#1827

### CODE REVIEW

**Comment by andrew-wzj** [comment]
Display course description in pop up window, and there is overflow issue for add to calendar button

**Comment by mla04762** [comment]
The file in `src/views/Home/components/CLWCreditsDaysLeft/index.jsx` has the variables `firstDay` and `endDay`. I think we could use the way these are set to get the startDate and endDate for the semester based on the current session.

File: src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx [hunk 1]
```diff
@@ -38,8 +37,11 @@ const ScheduleDialog = (props) => {
             <>
               <add-to-calendar-button
                 name={props.selectedCourseInfo.title}
-                startDate={format(new Date(props.selectedCourseInfo.start), 'yyyy-MM-dd')}
-                //endDate={format(new Date(props.selectedCourseInfo.end), 'yyyy-MM-dd')}
+                startDate={format(
+                  setDay(new Date(props.selectedCourseInfo.start), 4),
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx:41]:
```suggestion
                  setDay(new Date(props.selectedCourseInfo.start), 4),
```
why are we still hard-coding 4?

File: src/components/Profile/components/SchedulePanel/index.jsx [hunk 2]
```diff
@@ -153,8 +162,10 @@ const GordonSchedulePanel = (props) => {
   );
 
   const meetingDayArray = selectedCourseInfo?.meetingDays;
-  const recurringDays = meetingDayArray.map((day) => `${day}`).join(',');
-  console.log({ recurringDays });
+  const asdf = selectedCourseInfo;
+  const recurringDays = meetingDayArray?.map((day) => `${day}`).join(',');
+  console.log({ firstDay });
+  console.log({ lastDay });
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/index.jsx:168]:
```suggestion
  console.log({ firstDay });
  console.log({ lastDay });
```
remove console logs

File: src/services/schedule.ts [hunk 3]
```diff
@@ -63,6 +63,10 @@ function getMeetingDays(course: CourseSchedule): string[] {
   return dayArray;
 }
 
+const asdf = moment('2022-06-10T00:00:00');
+const month = asdf.month();
+console.log(month);
```
Comment (amos-cha) [inline src/services/schedule.ts:68]:
```suggestion
const asdf = moment('2022-06-10T00:00:00');
const month = asdf.month();
console.log(month);
```
im assuming this is testing. remember to remove it

File: src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx (hunk 1 above)
Comment (mla04762) [reply src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx:41]: We haven't pushed the commit with the changes yet, I will make sure to change this before review

File: src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+              " hh:mm aaaaa'm' ",
+            )}
+          </Typography>
+          <Typography className={styles.dialogTextMedium} align="left">
+            Week Day(s): {props.recurringDays}
+          </Typography>
+          <Typography className={styles.dialogTextMedium} align="left">
+            Term Date: {format(new Date(props.firstDay), 'yyyy-MM-dd')} to
+            {format(new Date(props.lastDay), ' yyyy-MM-dd')}
+          </Typography>
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx:48]: as above

File: src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                    // and endTime as null if they are all day events.
+                    startTime={
+                      props.selectedCourseInfo.allDay
+                        ? null
+                        : format(new Date(props.selectedCourseInfo?.start), 'HH:mm')
+                    }
+                    endTime={
+                      props.selectedCourseInfo.allDay
+                        ? null
+                        : format(new Date(props.selectedCourseInfo?.end), 'HH:mm')
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx:81]: would be nice to have a formatter

File: src/components/Profile/components/SchedulePanel/index.jsx [hunk 6]
```diff
@@ -138,6 +165,14 @@ const GordonSchedulePanel = (props) => {
     </div>
   );
 
+  // We had to make the meeting day array in to a string with commas because the add-to-calendar
+  // required a string of weekdays for setting the recurring days
+
+  const meetingDayArray = selectedCourseInfo?.meetingDays;
+  const recurringDays = meetingDayArray?.map((day) => `${day}`).join(', ');
+  const courseTitle = selectedCourseInfo?.title.split('in')[0];
+  const courseLocation = selectedCourseInfo?.title.split('in')[1];
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/index.jsx:174]: this is not particularly good practice. Why is this not handled when courses are selected?

**Review by amos-cha — CHANGES_REQUESTED** [review]

File: src/components/Profile/components/SchedulePanel/components/ScheduleCalendar/index.jsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
```
[the remark this answers is not in the document]
Comment (mla04762) [reply src/components/Profile/components/SchedulePanel/components/ScheduleCalendar/index.jsx:1]: This was for another method that we were trying to use, I got rid of the import.

File: src/components/Profile/components/SchedulePanel/components/ScheduleCalendar/index.jsx [hunk 8]
```diff
@@ -62,7 +62,7 @@ const GordonScheduleCalendar = (props) => {
   } else {
     let Resource = ({ localizer = momentLocalizer(Moment) }) => (
       <Calendar
-        selectable={props.isOnline}
+        selectable={true}
```
[the remark this answers is not in the document]
Comment (mla04762) [reply src/components/Profile/components/SchedulePanel/components/ScheduleCalendar/index.jsx:65]: This was for a temporary test that was supposed to be removed, it is reverted now.

File: src/components/Profile/components/SchedulePanel/components/ScheduleCalendar/index.jsx [hunk 9]
```diff
@@ -71,12 +71,14 @@ const GordonScheduleCalendar = (props) => {
...
         view={['day']}
-        onDoubleClickEvent={(event) => props.handleDoubleClick(event)}
+        //onDoubleClickEvent={(event) => props.handleDoubleClick(event)}
         defaultDate={Moment(new Date())}
         resources={resourceMap}
         resourceIdAccessor="resourceId"
         resourceTitleAccessor="resourceTitle"
         formats={formats}
+        onSelectEvent={props.onSelectEvent}
+        onSelecting={(slot) => false}
```
[the remark this answers is not in the document]
Comment (mla04762) [reply src/components/Profile/components/SchedulePanel/components/ScheduleCalendar/index.jsx:81]: This is basically disabling the drag select on the schedule panel which is no longer needed anymore.

File: src/components/Profile/components/SchedulePanel/ScheduleHeader.module.scss [hunk 10]
```diff
@@ -25,6 +25,7 @@
   }
 }
 
+.officeHourTitle {
 .officeHourTitle {
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/ScheduleHeader.module.scss]: duplicate?

File: src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+                Week Day(s): {props.recurringDays}
+              </Typography>
+              <Typography className={styles.dialogTextMedium} align="left">
+                Term Date: {formatter(props.firstDay, 'yyyy-MM-dd')} to
+                {formatter(props.lastDay, ' yyyy-MM-dd')}
+              </Typography>
+            </DialogContent>
+          </>
+        )}
+        <DialogActions style={{ overflow: 'hidden', flexDirection: 'column' }}>
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx:55]: css

File: src/services/schedule.ts [hunk 12]
```diff
@@ -23,9 +23,12 @@ type ScheduleEvent = {
   title: string;
   start: Date;
   end: Date;
-  resourceId: number;
+  resourceId: string;
+  meetingDays: string[];
 };
 
+type dayMap = {};
```
Comment (amos-cha) [inline src/services/schedule.ts:30]:
```suggestion
```

**Review by amos-cha — CHANGES_REQUESTED** [review]

File: src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+import styles from './ScheduleDialog.module.css';
+
+const dayArr = ['MO', 'TU', 'WE', 'TH', 'FR', 'SA'];
+
+const formatter = (date, display, isAllDay) => {
+  if (isAllDay) {
+    return null;
+  } else {
+    return format(new Date(date), display);
+  }
```
Comment (amos-cha) [inline src/components/Profile/components/SchedulePanel/components/ScheduleDialog/index.jsx:21]:
```suggestion
    if (isAllDay) 
      return null;
  return format(new Date(date), display);
  }
```

**Review by amos-cha — APPROVED** [review]

**Pull request merged**